### ISSUE TITLE
recv_into(…, MSG_DONTWAIT) behaviour

### ISSUE THREAD

**Issue by svinota** [eventlet/eventlet#323]
The eventlet library changes the behaviour of the  `socket.recv_into(…, MSG_DONTWAIT)` call.

* Python version: 2.7.11
* Eventlet version: python2-eventlet-0.17.4-4.fc23.noarch (but is reproduced on other platforms as well)
* Reproducible: always
* Expected result: socket.error: [Errno 11] Resource temporarily unavailable
* Real result: the call hangs

```
$ python
Python 2.7.11 (default, Mar 31 2016, 20:46:51) 
[GCC 5.3.1 20151207 (Red Hat 5.3.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import eventlet
>>> from socket import *
>>> eventlet.monkey_patch()
>>> s = socket(AF_NETLINK, SOCK_DGRAM)
>>> s.recv_into(bytearray((0, 0)), 2, MSG_DONTWAIT)
```

**Comment by svinota** [comment]
The same for `recv()`, so I'm changing the topic.

Looks like eventlet doesn't respect MSG_DONTWAIT and calls poll() here?

**Comment by temoto** [comment]
Yes, Eventlet was specifically designed to glue performance of non-blocking with straight readable code. It would seem possible to add handling of those flags, but so far I don't see a valid use case. Please show the code.

**Comment by svinota** [comment]
Since the beginning of June a lot of things passed by.

No, I will not show the code, since already solved the task in a different way. Nevertheless, I would insist that it is an issue, not an enhancement: the library simply breaks the Python stdlib functionality, and thus any code that would use MSG_DONTWAIT.

If there is no possibility to support the standard, at least it would be nice to warn the user that the behaviour of the socket operations was changed to opposite in the case of MSG_DONTWAIT, but not silently ignoring the flag.

Any valid use case for MSG_DONTWAIT is just the very case it was invented for, e.g. dependent socket operations in a single-threaded process.
